common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -693,6 +693,9 @@ public ICacheRecord load(@NonNull final String clientId,
                 account.getAuthorityType()
         );
 
+        // 'Preloading' our credentials to avoid repeated expensive cache hits
+        final List<Credential> allCredentials = mAccountCredentialCache.getCredentials();
+
         // Load the AccessTokens
         final List<Credential> accessTokens = mAccountCredentialCache.getCredentialsFilteredBy(
```

Comment:
>mAccountCredentialCache.getCredentialsFilteredBy [](http://example.com/codeflow?start=46&length=48)

nit: we might as well create a function that can return all the tokens.. in one call.. mAccountCredentialCache.getCredentials(List<CredentialTypes>, OtherfilteringCrieteria)

Reply:
I think we already have one.. 

getCredentialsFilteredBy(@Nullable final String homeAccountId,
                                                     @Nullable final String environment,
                                                     @NonNull final Set<CredentialType> credentialTypes,
                                                     @Nullable final String clientId,
                                                     @Nullable final String realm,
                                                     @Nullable final String target,
                                                     @Nullable final String authScheme,
                                                     @Nullable final String requestedClaims) {

Reply:
ah.. its also internally doing the same ting in a for loop... nvm